### PR TITLE
Feature/eui 1150 proxy tunnel and s2s refactor

### PR DESCRIPTION
**Pull request by paddybasi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-1150

### Change description ###
Implemented generic tunnel, we can now use the app with node versions 10+. Also cleaned up config code and s2s refactor.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: server/api/middlewares/validateS2Stoken.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { Request, Response, NextFunction } from 'express';
+import config from 'config';
+import { ERROR_CLIENT_NOT_WHITELISTED, ERROR_NO_S2S_TOKEN } from '../errors';
+import { getTokenDetails } from '../services/tokenDetails.service';
+
+// Check if S2S token contains 'Bearer', if not , add it.
+export const formatS2SToken = (token: string): string => {
```
Comment (andywilkinshmcts) [inline server/api/middlewares/validateS2Stoken.ts]: This seems wrong. As I understand it an s2s token shouldn't start with bearer

File: server/common/server.ts [hunk 2]
```diff
@@ -17,10 +18,11 @@ export default class ExpressServer {
...
         const sessionSecret = 'secret';
         const root = path.normalize(__dirname + '/../..');
+        tunnel.init();
         app.set('appPath', root + 'client');
-        app.use(bodyParser.json({ limit: requestLimit || '100kb' }));
-        app.use(bodyParser.urlencoded({ extended: true, limit: requestLimit || '100kb' }));
-        app.use(bodyParser.text({ limit: requestLimit || '100kb' }));
+        app.use(bodyParser.json({limit: requestLimit || '100kb'}));
+        app.use(bodyParser.urlencoded({extended: true, limit: requestLimit || '100kb'}));
+        app.use(bodyParser.text({limit: requestLimit || '100kb'}));
```
Comment (andywilkinshmcts) [inline server/common/server.ts]: I know that this hasn't changed but some of the CCD api requests return more that 100K of data. How will this setting work with that

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: server/api/middlewares/validateS2Stoken.ts (hunk 1 above)
Comment (paddybasi) [reply server/api/middlewares/validateS2Stoken.ts]: This is just a refactor of the S2S branch that @borislavpetrovhmcts created. But I do believe the rpe service does fail if bearer is not passed in.

File: server/common/server.ts (hunk 2 above)
Comment (paddybasi) [reply server/common/server.ts]: Yes I agree. Again I didn't explicitly code this part, I only added in the tunnel.init(). However I believe the intention was to abstract these out into config values, but have sensible defaults (the scaffolding tool provided the defaults)

**Pull request closed without merging**